src/templates/common/encoding/article_jats_1_2.xml

```diff
@@ -33,52 +36,62 @@
                 <article-title>{{ article.title }}</article-title>
             </title-group>
             <contrib-group>
-            {% for author in article.frozen_authors.all %}
-                <contrib contrib-type="author">
-                    {% if author.orcid %}
-                    <contrib-id contrib-id-type="orcid">{{ author.orcid }}</contrib-id>
-                    {% endif %}
-                    <name>
-                        <surname>{{ author.last_name }}</surname>
-                        <given-names>{{ author.given_names }}</given-names>
-                    </name>
-                    {% if author.is_correspondence_author %}
-                    <email>{{ author.author.email }}</email>
-                    {% endif %}
-                    {% if article.journal|setting:'use_credit' and author.credits.exists %}
-                        {% for credit_record in author.credits %}
-                            <role
-                                vocab="credit"
-                                vocab-identifier="https://credit.niso.org/"
-                                vocab-term="{{ credit_record.get_role_display }}"
-                                vocab-term-identifier="{{ credit_record.uri }}">
-                                {{ credit_record.get_role_display }}
-                            </role>
-                        {% endfor %}
-                    {% endif %}
-                    <xref ref-type="aff" rid="aff-{{ forloop.counter }}"/>
-                </contrib>
-            {% endfor %}
+                {% for item in article.frozen_authors_for_jats_contribs %}
+                    {% with author=item.author affiliations=item.affiliations %}
+                        <contrib contrib-type="author"{% if author.is_correspondence_author %} corresp="yes"{% endif %}>
+                            {% if author.orcid %}
```

Comment:
call me superstitious, but shall we check `author.orcid_uri` here to avoid having different results from `author.orcid` and `author.orcid_uri`? Seems like if the orcid was malformed, `author.orcid_uri` will return a falsey value here

Reply:
Yeah, a good shout.